src/test/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLoggerTest.java

```diff
@@ -126,7 +126,8 @@ void whenSetDimension_withInvalidValue_thenThrowInvalidDimensionException(
 
     @Test
     void whenSetDimension_withNameTooLong_thenThrowDimensionException() {
-        String dimensionName = "a".repeat(Constants.MAX_DIMENSION_NAME_LENGTH + 1);
+        String dimensionName =
+                new String(new char[Constants.MAX_DIMENSION_NAME_LENGTH + 1]).replace("\0", "a");
```

Comment:
instead of replacing, you can use `Arrays.fill()`
https://docs.oracle.com/javase/6/docs/api/java/util/Arrays.html#fill%28int%5B%5D,%20int%29